File: clojure/sum-of-multiples/src/sum_of_multiples.clj
(ns sum-of-multiples)

(defn- any-multiple? [i nums]
  (some #(zero? (rem i %)) nums))

(defn sum-of-multiples
  ([n] (sum-of-multiples [3 5] n))
  ([nums n]
   (apply + (filter #(any-multiple? % nums) (range 1 n)))))

File: go/sum-of-multiples/sum_of_multiples.go
package summultiples

func MultipleSummer(xs ...int) func(int) int {
	return func(limit int) int {
		sum := 0
		for i := 0; i < limit; i++ {
			if isMultipleOfAny(i, xs) {
				sum += i
			}
		}
		return sum
	}
}

func isMultipleOfAny(i int, xs []int) bool {
	for _, x := range xs {
		if i%x == 0 {
			return true
		}
	}
	return false
}
